Cannot drag and drop into the HyperTreeList

Hello.

I want to use the drag&drop feature into the HyperTreeList. I get the same errors I get when using the HyperTreeList demo in the package wxPython-demo-4.0.0a2.dev3038+953a2e5.
I am using Phoenix 4 and Python 3.5 in Windows 10 native on a i5@2,2GHz with 8GB of RAM.
Having checked the EVT_TREE_BEGIN_DRAG and EVT_TREE_END_DRAG events , when I attempt to drag a simple item from one node to another  I get the following errors:
Traceback (most recent call last):
  File "C:\TOOLS\python\python35\lib\site-packages\wx\lib\agw\hypertreelist.py", line 3528, in OnMouse
    self._dragImage.BeginDrag(wx.Point(0,0), self)
wx._core.wxAssertionError: C++ assertion "Assert failure" failed at ..\..\src\common\wincmn.cpp(3346) in DoNotifyWindowAboutCaptureLost(): window that captured the mouse didn't process wxEVT_MOUSE_CAPTURE_LOST
Traceback (most recent call last):
  File "C:\TOOLS\python\python35\lib\site-packages\wx\lib\agw\hypertreelist.py", line 3608, in OnMouse
    self.ReleaseMouse()
wx._core.wxAssertionError: C++ assertion "!wxMouseCapture::stack.empty()" failed at ..\..\src\common\wincmn.cpp(3319) in wxWindowBase::ReleaseMouse(): Releasing mouse capture but capture stack empty?
"
Googling I found out the wx.lib.agw.aui.AuiNotebook had a very similar issue. Some possible solution (see https://github.com/wxWidgets/Phoenix/pull/321 ) involved moving the self.HasCapture/self.ReleaseMouse after the self._dragImage.EndDrag() in the OnMouse handler.
By doing that I got rid of the "Releasing mouse capture but capture stack empty?" error message but I was left with the wxEVT_MOUSE_CAPTURE_LOST error message.
I am not sure how to proceed...
Are you aware of this issue? Do you have any idea on how to solve it?

Thank you.

--- a/wx/lib/agw/hypertreelist.py
+++ b/wx/lib/agw/hypertreelist.py
@@ -1076,7 +1076,8 @@ class TreeListHeaderWindow(wx.Window):
             if event.LeftDown() or event.RightUp():
                 if hit_border and event.LeftDown():
                     self._isDragging = True
-                    self.CaptureMouse()
+                    if not self.HasCapture():
+                        self.CaptureMouse()
                     self._currentX = x
                     self.DrawCurrent()
                     self.SendListEvent(wx.wxEVT_COMMAND_LIST_COL_BEGIN_DRAG, event.GetPosition())
@@ -2102,6 +2103,7 @@ class TreeListMainWindow(CustomTreeCtrl):
         self.Bind(wx.EVT_PAINT, self.OnPaint)
         self.Bind(wx.EVT_MOUSE_EVENTS, self.OnMouse)
         self.Bind(wx.EVT_SCROLLWIN, self.OnScroll)
+        self.Bind(wx.EVT_MOUSE_CAPTURE_LOST, lambda evt: None)
 
         # Sets the focus to ourselves: this is useful if you have items
         # with associated widgets.
@@ -3582,7 +3584,8 @@ class TreeListMainWindow(CustomTreeCtrl):
 
                 # we're going to drag this item
                 self._isDragging = True
-                self.CaptureMouse()
+                if not self.HasCapture():
+                    self.CaptureMouse()
                 self.RefreshSelected()
 
                 # in a single selection control, hide the selection temporarily
@@ -3603,8 +3606,8 @@ class TreeListMainWindow(CustomTreeCtrl):
             # end dragging
             self._dragCount = 0
             self._isDragging = False
-            if self.HasCapture():
-                self.ReleaseMouse()
+#            if self.HasCapture():
+#                self.ReleaseMouse()
             self.RefreshSelected()
 
             # send drag end event event

